### PR TITLE
Import old attestation store

### PR DESCRIPTION
**Pull request by shayzluf**
Feature

**What does this PR do? Why is it needed?**
Data migration function from old attestation protection db data structure to new one
**Which issues(s) does this PR fix?**

Part of #7196

**Other notes for review**

### CODE REVIEW

File: validator/db/kv/attestation_history_new.go [hunk 1]
```diff
@@ -82,7 +83,7 @@ func (hd EncHistoryData) getTargetData(ctx context.Context, target uint64) (*His
 
 	history.Source = bytesutil.FromBytes8(hd[cursor : cursor+sourceSize])
 	sr := make([]byte, 32)
-	copy(hd[cursor+sourceSize:cursor+historySize], sr)
+	copy(sr, hd[cursor+sourceSize:cursor+historySize])
```
Comment (nisdas) [inline validator/db/kv/attestation_history_new.go:86]: why is this inverted ?

File: validator/db/kv/attestation_history_new.go [hunk 2]
```diff
@@ -150,3 +151,56 @@ func (store *Store) SaveAttestationHistoryNewForPubKeys(ctx context.Context, his
...
+		}); err != nil {
+			return errors.Wrapf(err, "could not retrieve attestations for source in %s", store.databasePath)
+		}
+
+		return nil
+	}); err != nil {
+		return err
+	}
+	allKeys = removeDuplicateKeys(allKeys)
+	var keys [][48]byte
```
Comment (nisdas) [inline validator/db/kv/attestation_history_new.go:177]: if you need a variable sized array of [48] bytes, you should just do it above. You again copy it there and copy the pubkey again here.

Comment (shayzluf) [reply validator/db/kv/attestation_history_new.go:177]: `removeDuplicateKeys` uses [][]byte

File: validator/db/kv/attestation_history_new.go (hunk 1 above)
Comment (shayzluf) [reply validator/db/kv/attestation_history_new.go:86]: first var is destination, it was a bug. discovered by import data test

Comment (shayzluf) [reply validator/db/kv/attestation_history_new.go:86]: added this to the get target data test

File: validator/db/kv/attestation_history_new.go (hunk 2 above)
Comment (nisdas) [reply validator/db/kv/attestation_history_new.go:177]: not sure why that is the type used by a method called `removeDuplicateKeys` . All BLS pubkeys are 48 bytes, there is no variation in their size. If it is an easy change, we should do it if not you can just leave it as it is.

Comment (shayzluf) [reply validator/db/kv/attestation_history_new.go]: fixed

File: validator/db/kv/manage.go [hunk 3]
```diff
@@ -305,13 +305,13 @@ func addAttestations(bucket *bolt.Bucket, attestations pubKeyAttestations) error
...
 
-func removeDuplicateKeys(keys [][]byte) [][]byte {
+func removeDuplicateKeys(keys [][48]byte) [][48]byte {
 	last := 0
 
 next:
 	for _, k1 := range keys {
 		for _, k2 := range keys[:last] {
-			if bytes.Equal(k1, k2) {
+			if bytes.Equal(k1[:], k2[:]) {
```
Comment (nisdas) [inline validator/db/kv/manage.go:314]: if its a fixed sized array you can compare it directly.

Comment (nisdas) [inline validator/db/kv/manage.go:314]:
```suggestion
			if k1 == k2 {
```

Comment (shayzluf) [reply validator/db/kv/manage.go:314]: thanks

**Review by nisdas — APPROVED** [review]

**Review by nisdas — APPROVED** [review]

**Pull request merged**